Add Server Mono

caidanw opened the pull request:
#### Description

Added the font Server Mono, followed the `contributing.md` guide.

#### Requirements / Checklist

- [x] Read the [Contributing Guidelines](https://github.com/ryanoasis/nerd-fonts/blob/-/contributing.md)
- [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan.
      Issue number where discussion took place: #xxx
- [x] If this contains a font/glyph add its origin as background info below (e.g. URL)
- [x] Verified the license of any newly added font, glyph, or glyph set. License is: SIL Open Font License 1.1

#### What does this Pull Request (PR) do?

Add the Server Mono font

#### How should this be manually tested?

#### Any background context you can provide?

Origin: https://github.com/internet-development/www-server-mono

I'm the maintainer of this font, please let me know if there are any questions/concerns and I'd be happy to help.

#### What are the relevant tickets (if any)?

#### Screenshots (if appropriate or helpful)

Finii commented on src/unpatched-fonts/ServerMono/config.cfg:
```diff
@@ -0,0 +1,4 @@
+# These config files are read by the font-patcher
+# They are INI style files, but some keys have JSON values
+[Config]
+commandline: --has-no-italic --boxdrawing --complete --adjust-line-height
```
### `has-no-italic`

I fear that is not working, because it means in fact `has-oblique-instead-of-italic`, but here we have _Slanted_.

This needs a change in the name-parser (which is not a problem).

## `boxdrawing`

How strange is that, they have empty glyphs there? :sob: (Instead of no glyph).
I'd call that a bug in the source font.
:+1:

![image](https://github.com/user-attachments/assets/18a50abc-3e5d-441b-a110-4003fe5d9ee9)

##  `complete`

That is the default

## `adjust-line-height`

Any reason for this?
I believe it is not changing anything and hunt for people who see a difference?

But here, the 'cell' height is already even, so it does not even do anything.

![image](https://github.com/user-attachments/assets/c0aec06b-7fa9-4e75-8f1b-e590a2c7ecc3)

caidanw replied:
Thank you for this, this incredibly helpful to understand the options.
I'll remove the default `complete` and useless `adjust-line-height`.

Yeah I am not sure what happened with the empty box drawing glyphs for the slanted style. I know our box drawing glyphs are slightly wrong in position by a point or two, so I opted to just replace them in their entirety with the option `boxdrawing`.

I'm fixing the font now to export under `Oblique` instead of `Slanted`, I'll update this PR with the new files.

Finii commented on bin/scripts/lib/fonts.json:
```diff
@@ -887,6 +887,21 @@
+      "unpatchedName": "Server Mono",
+      "licenseId": "OFL-1.1-no-RFN",
+      "RFN": false,
+      "version": "0.0.8",
```
They put `1.0.0` resp. `1.000` into the font file, so that is an upstream bug :unamused:

Finii reviewed (commented):
You requested a review ;-)

The `fontfilenames` file has been forgotten (but I also hate that the information is scattered so much), probably we should autogenerate that file instead.

(No need for you to do anything, just noticing.)

https://github.com/ryanoasis/nerd-fonts/tree/master/src/unpatched-fonts#regenerating-the-fontfilenames-file

Finii commented on bin/scripts/lib/fonts.json:
```diff
@@ -887,6 +887,21 @@
+      "folderName": "ServerMono",
+      "imagePreviewFont": "ServerMono Nerd Font",
+      "imagePreviewFontSource": "ServerMono/ServerMono-Regular.otf",
+      "linkPreviewFont": "server-mono",
```
That is the link addendum to https://programmingfonts.org

caidanw replied:
Ah got it thanks! Will update!

caidanw commented:
Thanks for taking the time to review this and help guide me on the stuff I missed! I really appreciate the effort and knowledge! I agree that it'd be helpful for new folks like myself to have the documentation in one place (or at least pointing to the various places).

---

I noticed a bug when generating the `fontfilenames` file, I think the script may have been written with an older `jq` version because the `.fonts[$i]` gave an error. I slightly modified the script to make it work:
```diff
diff --git a/src/unpatched-fonts/README.md b/src/unpatched-fonts/README.md
index be50bdb4..4b597e19 100644
--- a/src/unpatched-fonts/README.md
+++ b/src/unpatched-fonts/README.md
@@ -13,8 +13,8 @@ all the fonts in `fontfilenames`. This can be used via
 
 Just call
 
-        jq -r '.fonts[$i].imagePreviewFontSource' ../../bin/scripts/lib/fonts.json \
+        jq -r '.fonts[].imagePreviewFontSource' ../../bin/scripts/lib/fonts.json \
             | grep -v '\.sfd$' \
             | sed 's/ /\\ /g' \
             | LC_ALL=C sort --ignore-case \
-            > fontfilenames
+            >| fontfilenames

```
I removed the problematic `$i` as it was an undeclared jq variable, and I changed the redirection to force overwrite even when folks like myself have `noclobber` enabled by default.

I can add this change to this PR, but if you'd prefer to keep it separate then I'll let you add it on your own accord.
